### PR TITLE
check makefiles (check_source.sh)

### PR DESCRIPTION
**Pull request by karel-m**

### CODE REVIEW

**Comment by karel-m** [comment]
Seems to be working - https://travis-ci.org/libtom/libtomcrypt/jobs/206910033

File: check_source.sh [hunk 1]
```diff
@@ -10,6 +10,19 @@ if [ -f check-source.pl ] ; then
...
+if [ -d .git ] ; then
+  echo "checking makefiles..."
+  ./updatemakes.sh
+  outofdate="$(git diff --name-only HEAD)"
+  if [ -n "$outofdate" ] ; then
+    echo "FAIL: makefiles not updated"
+    echo "$outofdate";
+    exit 1
+  else
+     echo "PASS"
```
Comment (sjaeckel) [inline check_source.sh]: indentation

File: check_source.sh [hunk 2]
```diff
@@ -10,6 +10,19 @@ if [ -f check-source.pl ] ; then
   perl check-source.pl || exit 1
 fi
 
+if [ -d .git ] ; then
+  echo "checking makefiles..."
+  ./updatemakes.sh
+  outofdate="$(git diff --name-only HEAD)"
```
Comment (sjaeckel) [inline check_source.sh]: this makes the entire flow dependent if it's run from the git repo

as we're going to distribute archives as release, I think this needs another solution, but the idea is good!

File: makefile [hunk 3]
```diff
@@ -42,7 +42,7 @@ LIBTEST_S=$(LIBTEST)
 
 #List of objects to compile.
 #START_INS
-OBJECTS=src/ciphers/aes/aes_enc.o src/ciphers/aes/aes.o src/ciphers/anubis.o src/ciphers/blowfish.o \
+OBJECTS=src/ciphers/aes/aes.o src/ciphers/aes/aes_enc.o src/ciphers/anubis.o src/ciphers/blowfish.o \
```
Comment (sjaeckel) [inline makefile]: srsly, the new sorting sorts alphabetically in a different order than the previous sorting? :rofl:

**Comment by karel-m** [comment]
> srsly, the new sorting sorts alphabetically in a different order than the previous sorting? 

yes, because now we rely on perl's built-in sort (which should work more or less the same on all platforms), the old version was based on `sort` command which depends on locale you have set in your shell

**Comment by karel-m** [comment]
is updated version fine for merge?

File: check_source.sh [hunk 4]
```diff
@@ -10,6 +10,19 @@ if [ -f check-source.pl ] ; then
   perl check-source.pl || exit 1
 fi
 
+if [ -f ./updatemakes.sh ] ; then
+  echo "checking makefiles..."
+  beforeupdate=$(grep '' makefile*)
+  ./updatemakes.sh
```
Comment (sjaeckel) [inline check_source.sh]: I like that a lot better, but I still don't like the fact that the files get now modified on disk when you run this (I especially dislike that the filename lies and says 'check' and then does 'modify' :-) )

**Comment by karel-m** [comment]
now completely new approach

File: updatemakes.sh [hunk 5]
```diff
@@ -1,26 +1,5 @@
...
-perl filter.pl makefile.msvc tmplist
-sed -e 's/\.o /.obj /g' -e 's/ *$//' < tmp.delme > makefile.msvc
-rm -f tmp.delme
-
-rm -f tmplist
-
-if [ $# -eq 1 ] && [ "$1" == "-c" ]; then
-  git add ${makefiles[@]} makefile.msvc && git commit -m 'Update makefiles'
-fi
+exit 0
```
Comment (sjaeckel) [inline updatemakes.sh]: missing newline at eof

**Comment by sjaeckel** [comment]
regarding 7046af108e90df65942c92f417594e2cefdf9464: I think we should provide aes_enc for compatibility reasons, or not?

**Comment by karel-m** [comment]
> regarding 7046af1: I think we should provide aes_enc for compatibility reasons, or not?

As for MS Windows builds via `*.vcproj` I vote leaving out `aes_anc`. 

And generally I would prefer not using the hack we currently use for building `aes.o` + `aes_enc.o` from a single `aes.c`. IMO having a separate `aes_enc.c` + a handy set of includes+ifdefs would be much better and will make maintenance easier. But it is perhaps worth a separate ticket.

Do you (or any other reader of this ticket) have any idea what was the motivation for `aes_anc` and why only AES and not generally other ciphers as well?

**Comment by karel-m** [comment]
I have returned the aes_enc hack to helper.pl - 6f9e427f77 (it is commented out so that it can be easily put back if needed)

**Pull request merged**